devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -97,7 +98,12 @@ protected void doExecute() throws MojoExecutionException {
                         final Path standardJar = curatedApplication.getAppModel().getAppArtifact().getPaths().getSinglePath();
                         if (Files.exists(standardJar)) {
                             try {
-                                Files.deleteIfExists(result.getJar().getOriginalArtifact());
+                                if (Files.exists(result.getJar().getOriginalArtifact())) {
+                                    Files.walk(result.getJar().getOriginalArtifact())
+                                            .sorted(Comparator.reverseOrder())
+                                            .map(Path::toFile)
+                                            .forEach(File::delete);
+                                }
```

Comment:
Perhaps we should call https://github.com/quarkusio/quarkus/blob/master/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/util/IoUtils.java#L58 instead as this is what we've been using everywhere until now.

Reply:
Ok, thanks forpoinnter, I will update the PR